asserts/database.go

```diff
@@ -89,6 +89,27 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// RevisionError indicates a revision improperly used for an operation.
+type RevisionError struct {
+	Used, Current int
+}
+
+func (e *RevisionError) Error() string {
+	if e.Used < 0 {
+		return fmt.Sprintf("used assertion revision %d is unknown", e.Used)
+	}
+	if e.Current < 0 {
```

Comment:
I think it was deliberate to let one use Current == -1 to get  a message about Used being wrong/unknown